tgui/packages/tgui/interfaces/PaperSheet.js

```diff
@@ -0,0 +1,71 @@
+import { toTitleCase } from 'common/string';
+import { Box, Flex, Button, TextArea } from '../components';
+import { useBackend, useSharedState, useLocalState } from '../backend';
+import { Window } from '../layouts';
+// import marked from 'marked';
+import { marked } from '../components/marked/marked';
+import { isFalsy } from 'common/react';
+
+import { createLogger } from '../logging';
+
+const logger = createLogger('PaperSheet');
+
+
+export const PaperSheet = (props, context) => {
+  const { act, data } = useBackend(context);
+  // https://github.com/segment-boneyard/socrates/blob/master/libs/marked.js
+  // his is older code where the updated
+  const {
+    text = "",
+    edit_sheet,
+    paper_color="white",
+    pen_color="black",
+    ...rest
+  } = data;
+  const [marked_value, setMarked] = useLocalState(context, text);
+  const handleOnInput = (e, value) => {
+    // Need to fix cut and paste humm
+    setMarked(value);
+  };
+
+  const readSheet = marked_text => {
+    return (
+      <Box p="0px" textColor={pen_color} backgroundColor={pen_color}
+        width="auto" height={(window.innerHeight) + "px"}>
+        {marked(isFalsy(marked_text) ? '' : marked_text,
+          { breaks: true, smartypants: true })}
+      </Box>
+    );
+  };
+  const editSheet = () => {
+    return (
+      <Flex direction="column" justify="center">
+        <Flex.Item>
+          <Button.Confirm
+            content="FINISH"
+            onClick={() => act('save', { 'text': marked_value })}
+          />
+        </Flex.Item>
+        <Flex.Item>
+          <Flex>
+            <Flex.Item width="50%" >
+              <TextArea fluid height={(window.innerHeight-60) + "px"}
+                onInput={handleOnInput} />
+            </Flex.Item>
+            <Flex.Item width="50%">
+              {readSheet(marked_value)}
+            </Flex.Item>
+          </Flex>
+        </Flex.Item>
+      </Flex>
+    );
+  };
```

Comment:
Please extract these functions into separate components. It's very easy in tgui 3.0, and it will separate concerns much better.

Reply:
Yea, originally that's how it was set up but I was having problems getting state to work across components.